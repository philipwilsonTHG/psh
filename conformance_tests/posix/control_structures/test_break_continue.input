# Test break and continue statements
# PSH implements break/continue for loop control (v0.16.0 feature)

echo "Testing continue in for loop:"
for i in 1 2 3 4 5; do
    if [ "$i" = "3" ]; then
        echo "Skipping $i"
        continue
    fi
    echo "Processing $i"
done

echo "Testing break in for loop:"
for i in 1 2 3 4 5; do
    if [ "$i" = "4" ]; then
        echo "Breaking at $i"
        break
    fi
    echo "Processing $i"
done

echo "Testing continue in while loop:"
count=0
while [ "$count" -lt 5 ]; do
    count=$((count + 1))
    if [ "$count" = "3" ]; then
        echo "Skipping count $count"
        continue
    fi
    echo "Count: $count"
done

echo "Testing break in while loop:"
count=0
while true; do
    count=$((count + 1))
    if [ "$count" -gt 3 ]; then
        echo "Breaking at count $count"
        break
    fi
    echo "Count: $count"
done

echo "Testing nested loops with break/continue:"
for outer in 1 2; do
    echo "Outer: $outer"
    for inner in a b c; do
        if [ "$outer" = "1" ] && [ "$inner" = "b" ]; then
            echo "  Continuing inner at $inner"
            continue
        fi
        if [ "$outer" = "2" ] && [ "$inner" = "c" ]; then
            echo "  Breaking inner at $inner"
            break
        fi
        echo "  Inner: $inner"
    done
done