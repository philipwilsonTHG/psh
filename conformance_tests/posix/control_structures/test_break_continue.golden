Testing continue in for loop:
Processing 1
Processing 2
Skipping 3
Processing 4
Processing 5
Testing break in for loop:
Processing 1
Processing 2
Processing 3
Breaking at 4
Testing continue in while loop:
Count: 1
Count: 2
Skipping count 3
Count: 4
Count: 5
Testing break in while loop:
Count: 1
Count: 2
Count: 3
Breaking at count 4
Testing nested loops with break/continue:
Outer: 1
  Inner: a
  Continuing inner at b
  Inner: c
Outer: 2
  Inner: a
  Inner: b
  Breaking inner at c
